src/deckgl-layers/layer-utils/cpu-aggregator.js

```diff
@@ -0,0 +1,606 @@
+        },
+        weight: {
+          prop: 'getColorWeight',
+          updateTrigger: 'getColorWeight'
+        },
+        aggregation: {
+          prop: 'colorAggregation'
+        }
+      },
+      updater: getGetValue
+    },
+    {
+      key: 'getBins',
+      triggers: {
+        _filterData: {
+          prop: '_filterData',
+          updateTrigger: '_filterData'
+        }
+      },
+      updater: getDimensionSortedBins
+    },
+    {
+      key: 'getDomain',
+      triggers: {
+        lowerPercentile: {
+          prop: 'lowerPercentile'
+        },
+        upperPercentile: {
+          prop: 'upperPercentile'
+        },
+        scaleType: {prop: 'colorScaleType'}
+      },
+      updater: getDimensionValueDomain
+    },
+    {
+      key: 'getScaleFunc',
+      triggers: {
+        domain: {prop: 'colorDomain'},
+        range: {prop: 'colorRange'},
+        scaleType: {prop: 'colorScaleType'}
+      },
+      onSet: {
+        props: 'onSetColorDomain'
+      },
+      updater: getDimensionScale
+    }
+  ],
+  getSubLayerAccessor
+};
+
+export const defaultElevationDimension = {
+  key: 'elevation',
+  accessor: 'getElevation',
+  getPickingInfo: (dimensionState, cell) => {
+    const {sortedBins} = dimensionState;
+    const elevationValue =
+      sortedBins.binMap[cell.index] && sortedBins.binMap[cell.index].value;
+    return {elevationValue};
+  },
+  nullValue: -1,
+  updateSteps: [
+    {
+      key: 'getValue',
+      triggers: {
+        value: {
+          prop: 'getElevationValue',
+          updateTrigger: 'getElevationValue'
+        },
+        weight: {
+          prop: 'getElevationWeight',
+          updateTrigger: 'getElevationWeight'
+        },
+        aggregation: {
+          prop: 'elevationAggregation'
+        }
+      },
+      updater: getGetValue
+    },
+    {
+      key: 'getBins',
+      triggers: {
+        _filterData: {
+          prop: '_filterData',
+          updateTrigger: '_filterData'
+        }
+      },
+      updater: getDimensionSortedBins
+    },
+    {
+      key: 'getDomain',
+      triggers: {
+        lowerPercentile: {
+          prop: 'elevationLowerPercentile'
+        },
+        upperPercentile: {
+          prop: 'elevationUpperPercentile'
+        },
+        scaleType: {prop: 'elevationScaleType'}
+      },
+      updater: getDimensionValueDomain
+    },
+    {
+      key: 'getScaleFunc',
+      triggers: {
+        domain: {prop: 'elevationDomain'},
+        range: {prop: 'elevationRange'},
+        scaleType: {prop: 'elevationScaleType'}
+      },
+      onSet: {
+        props: 'onSetElevationDomain'
+      },
+      updater: getDimensionScale
+    }
+  ],
+  getSubLayerAccessor
+};
+
+export const defaultDimensions = [
+  defaultColorDimension,
+  defaultElevationDimension
+];
+
+export default class CPUAggregator {
+  constructor(opts = {}) {
+    this.state = {
+      layerData: {},
+      dimensions: {
+        // color: {
```

Comment:
are these lines used as reference? otherwise, can you delete them?

Reply:
reference